src/theory/quantifiers/sygus/example_eval_cache.cpp

```diff
@@ -51,17 +51,16 @@ Node ExampleEvalCache::addSearchVal(Node bv)
     return Node::null();
   }
   std::vector<Node> vals;
-  evaluateVec(bv, vals, false);
+  evaluateVec(bv, vals, true);
   Trace("sygus-pbe-debug") << "Add to trie..." << std::endl;
   Node ret = d_trie.addOrGetTerm(bv, vals);
   Trace("sygus-pbe-debug") << "...got " << ret << std::endl;
   // Only save the cache data if necessary: if the enumerated term
   // is redundant, its cached data will not be used later and thus should
   // be discarded.
-  if (ret == bv)
+  if (ret != bv)
   {
-    std::vector<Node>& eocv = d_exOutCache[bv];
-    eocv.insert(eocv.end(), vals.begin(), vals.end());
+    clearEvaluationCache(bv);
```

Comment:
Can you comment on why you had to revert this?

Reply:
It's in the PR description.  The issue is two-fold:
- If it was already cached prior to entering this function and is not redundant, we should not add to the cache again (we were doubling the size of  the example out vector, leading to segfaults),
- If it was already cached prior to entering this function and is redundant, we should clear the cache immediately.

Reply:
I see now. I think you should add these comments to the code. It's an insight that is hard to gain otherwise.